lib/opal/rails/template_handler.rb

```diff
@@ -23,7 +23,11 @@ def local_assigns
 
       def assigns
         <<-'RUBY'.strip
-          JSON.parse(@_assigns.to_json).map { |key, val| "@#{key} = #{val.inspect};" }.join
+          if Rails.application.config.opal.auto_assign_instance_variables?(params.slice(:controller, :action))
```

Comment:
The problem with that in TemplateHandler context I do not have access to params. params becomes available only when the string is evaluated in view context. 

That is why I had added the the check inside the evaluated string. 

I am exploring the code around this first time, but if there is an alternative approach please let me know.